### PR TITLE
Send resource along with span attributes and kind/status

### PR DESCRIPTION
**Pull request by valkum**
The stackdriver exporter is currently not sending the `Resource` key-values. (Similar to #880)
This PR merges the `Resource` and `Attributes` and caps them at 32. This limit is imposed by [Google](https://cloud.google.com/trace/docs/reference/v2/rpc/google.devtools.cloudtrace.v2#google.devtools.cloudtrace.v2.Span).
The `Resource` key-values take precedence over `Attributes`. 

Furthermore, this PR adds support for mapping [SpanKind](https://cloud.google.com/trace/docs/reference/v2/rpc/google.devtools.cloudtrace.v2#spankind) and [Status](https://cloud.google.com/trace/docs/reference/v2/rpc/google.rpc#google.rpc.Status)

### CODE REVIEW

**Comment by cijothomas** [comment]
@valkum Please sign the CLA agreement, which is a pre-req for any contributions to OpenTelemtry repos. If you have already signed, but the CI check is still red, I think its due to an ongoing issue with EasyCla bot, being investigated separately.

**Comment by valkum** [comment]
> @valkum Please sign the CLA agreement, which is a pre-req for any contributions to OpenTelemtry repos. If you have already signed, but the CI check is still red, I think its due to an ongoing issue with EasyCla bot, being investigated separately.

Currently waiting for our CLA Manager to sign. Might take a while.

File: opentelemetry-stackdriver/src/lib.rs [hunk 1]
```diff
@@ -717,10 +730,13 @@ impl From<EvictedHashMap> for Attributes {
...
                 Some((key.to_owned(), v.into()))
             })
+            .take(32)
             .collect();
+
         Attributes {
             attribute_map,
-            dropped_attributes_count,
+            dropped_attributes_count: dropped_attributes_count
+                + (num_resource_attributes + num_attributes).saturating_sub(32) as i32,
```
Comment (TommyCpp) [inline opentelemetry-stackdriver/src/lib.rs:739]: nit: can we make 32 a constant? Similar in L733

File: opentelemetry-stackdriver/src/lib.rs [hunk 2]
```diff
@@ -691,11 +696,19 @@ pub enum MonitoredResource {
...
-    fn from(attributes: EvictedHashMap) -> Self {
+impl From<(EvictedHashMap, &Resource)> for Attributes {
+    /// Combines `EvictedHashMap` and `Resource` attributes into a maximum of 32.
+    ///
+    /// The `Resource` takes precedence over the `EvictedHashMap` attributes.
+    fn from((attributes, resource): (EvictedHashMap, &Resource)) -> Self {
         let mut dropped_attributes_count: i32 = 0;
-        let attribute_map = attributes
+        let num_resource_attributes = dbg!(resource.len());
+        let num_attributes = dbg!(attributes.len());
```
Comment (TommyCpp) [inline opentelemetry-stackdriver/src/lib.rs:706]: why `dbg!` here?

**Review by TommyCpp — COMMENTED** [review]
The logic is clear to me. Some small nits. Also could you include an entry in CHANGELOG?

File: opentelemetry-stackdriver/src/lib.rs (hunk 1 above)
Comment (hdost) [reply opentelemetry-stackdriver/src/lib.rs]: Seems to have been done now.

File: opentelemetry-stackdriver/src/lib.rs (hunk 2 above)
Comment (hdost) [reply opentelemetry-stackdriver/src/lib.rs]: Seems good now.

**Comment by djc** [comment]
I'm having issues signing into their management dashboard to act as CLA manager. Filed an issue against the LF Jira.

**Pull request merged**